tests/python/test_parallel_range_for.py

```diff
@@ -1,7 +1,8 @@
 import taichi as ti
 
 
-@ti.all_archs
+# such small block_dim will cause grid_dim too large for OpenGL...
```

Comment:
IIRC, OpenGL supports grid stride loop, so is this still a problem?

Reply:
You mean, when grid_dim is large, we 'share' some of the grid_dim into stride-loop-size? That would be non-trivial, since stride-loop-size is determined at compile-time, while grid_dim must be determined at runtime, for future cross-compile support #1601.

Reply:
I think having `ti.stride_loop(n)` would be a good idea.

Reply:
I think grid stride loop is meant to be adaptive. For example, if we cap both `block_dim` and `grid_dim`, such that `block_dim * grid_dim < N` (`N` for number of items of an SNode), the kernel should still be able to handle all the `N` items?